src/ex14_js-context/task-01.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+function Calculator() {
+  this.result = 0;
+  this.add = function(num) {
+    if(num === undefined) return this;
+    if (!Number.isNaN(num)) this.result += num;
+    return this;
+  }
+  this.subtract = function(num) {
+    if(num === undefined) return this;
+    if (!Number.isNaN(num)) this.result -= num;
+    return this;
+  }
+  this.divide = function(num) {
+    if(num === undefined) return this;
+    if (!Number.isNaN(num) && num !== 0) this.result /= num;
+    return this;
+  }
+  this.multiply = function(num) {
+    if(num === undefined) return this;
+    if (!Number.isNaN(num)) this.result *= num;
+    return this;
+  }
+  this.setState = function(num) {
+    if(num === undefined) return this;
+    if (!Number.isNaN(num)) this.result = num;
+    return this;
+  }
+  this.fetchData = function(cb) {
```

Comment:
I think prefix 'fetch' in the name of function isn't enough for emulate server request behavior :))
what would u think about web API?..

Reply:
That's a good point😁 Fresh commit adds timer to similate server response lag.